lptree.pas

```diff
@@ -1114,9 +1114,22 @@ procedure TLapeTree_OpenArray.ClearCache;
 end;
 
 function TLapeTree_OpenArray.addValue(Val: TLapeTree_Base): Integer;
```

Comment:
Can we turn Val into `ExprBase` instead of `Base`, so you don't need to check for 2 subclasses specifically? I think that would be a lot cleaner.